20_beta_diversity.Rmd

```diff
@@ -75,16 +58,22 @@ demonstration we will analyse beta diversity in GlobalPatterns, and observe the
 variation between stool samples and those with a different origin.
 
 ```{r prep-tse}
-# Example data
+# Load mia and import sample dataset
+library(mia)
 data("GlobalPatterns", package = "mia")
-
-# Data matrix (features x samples)
 tse <- GlobalPatterns
 
-# some beta diversity metrics are usually applied to relative abundances
+# Beta diversity metrics like Bray-Curtis are often applied to relabundances
 tse <- transformAssay(tse,
+                      assay.type = "counts",
                       method = "relabundance")
 
+# Other metrics like Aitchison to clr-transformed data
+tse <- transformAssay(tse,
+                      assay.type = "relabundance",
+                      method = "clr",
+                      pseudocount = 1)
```

Comment:
This should not affect (?) since every value get the same pseudocount, but i the documentation we suggest to use smallest positive value --> min(mat[mat > 0])

--> could we add `pseudocount = TRUE` option for this, and give a message on what pseudocount was used? (adding smallest positive value requires user to fetch first the assay which is not optimal...)

Reply:
Good question. Indeed 1 is problematic for relabundances for instance.  I think the idea has been that the users must explicitly do this. But I am not sure how necessary that is really.

Ok we can go with automation.

Reply:
Often min/2 is used. To distinguish from min. 

However this would work better for relabundances than for integer counts.. so perhaps just min as the default pseudocount..